### ISSUE TITLE
Stack Experience is count double

### ISSUE THREAD

**Issue by kdmcser** [vcmi/vcmi#4897]
**Describe the bug**
his is the 10 level attributes of ancientBehemoth（hero A2 D2）
![image](https://github.com/user-attachments/assets/9d0e9ee5-1e4a-43c3-81a1-9f4674d8f903)

this is the stackExperience of ancientBehemoth
```
	"core:ancientBehemoth" :
	{
		"stackExperience" :
		[
			{
				"bonus" :
				{
					"type" : "PRIMARY_SKILL",
					"subtype" : "primarySkill.attack"
				},
				"values" : [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
			},	
			{
				"bonus" :
				{
					"type" : "PRIMARY_SKILL",
					"subtype" : "primarySkill.defence"
				},
				"values" : [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
			},
			{
				"bonus" :
				{
					"type" : "CREATURE_DAMAGE",
					"subtype" : "creatureDamageMax"
				},
				"values" : [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
			},
			{
				"bonus" :
				{
					"type" : "CREATURE_DAMAGE",
					"subtype" : "creatureDamageMin"
				},
				"values" : [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
			},
			{
				"bonus" :
				{
					"valueType" : "PERCENT_TO_BASE",
					"type" : "STACK_HEALTH"
				},
				"values" : [ 5, 10, 15, 20, 25, 30, 35, 40, 45, 50 ]
			},
			{
				"bonus" :
				{
					"type" : "STACKS_SPEED"
				},
				"values" : [ 0, 0, 2, 3, 3, 3, 3, 3, 3, 4 ]
			}
                    ...
		]
	}

```
attack is 39 - 2 = 37, 18 addtional attack is added. this value should be 10 , but the real is 9 * 2
health is 570, 90% more than 300,  this value should be 50% , but the real is 45% * 2
speed is 18, 9 addtional speed is added. this value should be 4 , but the real is 9(why?)
and there are two additional attack bounse.

1. It seems that vcmi first load data/crexpbon.txt in stackExperience mod, then load the creature json which contains stackExperience again. So All bonuses is loaded twice.

2. I don't know why pick the 9th values when creature in level 10, this is the second problem.

3. The third problem is why +9 speed. that should be 6 since 1 and 2.

this problem occurs in 1.6 daily build, windows and android.

**Comment by Olegmods** [comment]
> **Describe the bug** his is the 10 level attributes of ancientBehemoth（hero A2 D2） ![image](https://private-user-images.githubusercontent.com/8289682/384945399-9d0e9ee5-1e4a-43c3-81a1-9f4674d8f903.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzMzQ2NzcsIm5iZiI6MTczMTMzNDM3NywicGF0aCI6Ii84Mjg5NjgyLzM4NDk0NTM5OS05ZDBlOWVlNS0xZTRhLTQzYzMtODFhMS05ZjQ2NzRkOGY5MDMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTFUMTQxMjU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWQ2ZDY5Mjk5MmU3NjYxYjFjOTM3NzE3OTQ3ZWYxZGM2MGQyMGJjODM0OTRjOWRhYmJmOTZkNmM3NzIxMzQxYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Jwu-8iqFeuZMQ-f-1Q0gFLK11O7DoAJ5v_qErYoR_NM)
> 
> this is the stackExperience of ancientBehemoth
> 
> ```
> 	"core:ancientBehemoth" :
> 	{
> 		"stackExperience" :
> 		[
> 			{
> 				"bonus" :
> 				{
> 					"type" : "PRIMARY_SKILL",
> 					"subtype" : "primarySkill.attack"
> 				},
> 				"values" : [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
> 			},	
> 			{
> 				"bonus" :
> 				{
> 					"type" : "PRIMARY_SKILL",
> 					"subtype" : "primarySkill.defence"
> 				},
> 				"values" : [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
> 			},
> 			{
> 				"bonus" :
> 				{
> 					"type" : "CREATURE_DAMAGE",
> 					"subtype" : "creatureDamageMax"
> 				},
> 				"values" : [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
> 			},
> 			{
> 				"bonus" :
> 				{
> 					"type" : "CREATURE_DAMAGE",
> 					"subtype" : "creatureDamageMin"
> 				},
> 				"values" : [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
> 			},
> 			{
> 				"bonus" :
> 				{
> 					"valueType" : "PERCENT_TO_BASE",
> 					"type" : "STACK_HEALTH"
> 				},
> 				"values" : [ 5, 10, 15, 20, 25, 30, 35, 40, 45, 50 ]
> 			},
> 			{
> 				"bonus" :
> 				{
> 					"type" : "STACKS_SPEED"
> 				},
> 				"values" : [ 0, 0, 2, 3, 3, 3, 3, 3, 3, 4 ]
> 			}
>                     ...
> 		]
> 	}
> ```
> 
> attack is 39 - 2 = 37, 18 addtional attack is added. this value should be 10 , but the real is 9 * 2 health is 570, 90% more than 300, this value should be 50% , but the real is 45% * 2 speed is 18, 9 addtional speed is added. this value should be 4 , but the real is 9(why?) and there are two additional attack bounse.
> 
> 1. It seems that vcmi first load data/crexpbon.txt in stackExperience mod, then load the creature json which contains stackExperience again. So All bonuses is loaded twice.
> 2. I don't know why pick the 9th values when creature in level 10, this is the second problem.
> 3. The third problem is why +9 speed. that should be 6 since 1 and 2.
> 
> this problem occurs in 1.6 daily build, windows and android.

Did you try removing crexpbon.txt from the mod? Without it the following problems appear?

**Comment by kdmcser** [comment]
removing it will cause game crash. my temporary solve is remove all content of this file except the first line

**Comment by Olegmods** [comment]
> removing it will cause game crash. my temporary solve is remove all content of this file except the first line

Does it work correctly if you disable stack experience configs in mod.json?

**Comment by Olegmods** [comment]
I guess it's better make a PR that either removes stack experience configs for creatures, or changes crexpbon.txt.

**Comment by misiokles** [comment]
I remember when I wrote stack experience into json because .txt file was read wrong. But it was for 0.99 version. As far I remember only first lines of crbom.txt is read.

**Comment by kdmcser** [comment]
> > removing it will cause game crash. my temporary solve is remove all content of this file except the first line
> 
> Does it work correctly if you disable stack experience configs in mod.json?

of course. remove either stack experience config or  data/crexpbon.txt content is OK

**Comment by kdmcser** [comment]
> I guess it's better to make a PR that either removes stack experience configs for creatures, or changes crexpbon.txt.

I dont know which is better. Or we should remove parsing crexpbon.txt logic in VCMI code?

**Comment by kdmcser** [comment]
> I remember when I wrote stack experience into json because .txt file was read wrong. But it was for 0.99 version. As far I remember only first lines of crbom.txt is read.

so after this error is fixed, things go wrong.
